api/controller.py

```diff
@@ -143,10 +143,22 @@ def __init__(self, _db, lanes=None, testing=False):
                 sys.exit()
 
         self.testing = testing
+        self.site_configuration_last_update = (
+            Configuration.site_configuration_last_update(self._db, timeout=0)
+        )
         self.lane_descriptions = lanes
         self.setup_one_time_controllers()
         self.load_settings()
         
+    def reload_settings_if_changed(self):
+        """If the site configuration has been updated, reload the
+        CirculationManager's configuration from the database.
+        """
+        last_update = Configuration.site_configuration_last_update(self._db, timeout=0)
```

Comment:
That's fine, but this is the method that runs on every request, not initialization.